### PR TITLE
Compile-time optimization: disable backwards pass for handle_contact_pairs

### LINKED ISSUE #697: Slow startup time for examples due to MJWarp Compilation Time
`example_robot_anymal_d.py` takes approximately 160 seconds (2.5 mins) to load from cold (fresh install).

Time is currently dominated by repeated compilation of MuJoCo Warp kernels: 

```
Module mujoco_warp._src.collision_convex.ccd_kernel_builder 1d08d48 load on device 'cuda:0' took 13025.32 ms  (compiled)
Module mujoco_warp._src.collision_convex.ccd_kernel_builder f41ccf5 load on device 'cuda:0' took 20842.54 ms  (compiled)
Module mujoco_warp._src.collision_convex.ccd_kernel_builder a10ce81 load on device 'cuda:0' took 29339.83 ms  (compiled)
```

and 

```
Module mujoco_warp._src.forward 612d0c2 load on device 'cuda:0' took 1563.28 ms  (compiled)
Module mujoco_warp._src.forward cec2705 load on device 'cuda:0' took 1449.57 ms  (compiled)
Module mujoco_warp._src.forward 43c3857 load on device 'cuda:0' took 1332.24 ms  (compiled)
```

We should check that MuJoCo Warp kernels are being compiled as few times as necessary and use unique compilation modules where appropriate since it seems `ccd_kernel_builder` time is growing with each recompilation.

### PR DESCRIPTION
**Pull request by adenzler-nvidia**
Fixes #697 

The main remaining big chunk is the compilation of the backwards pass of the handle_contact_pairs kernel. Compile time goes from 40s down to 3s for the humanoid example, and the gains are similar for all the other robots.

This PR adds a generator function to en/disable the backwards pass of that kernel based on whether the model requests gradient storage during finalize. We can discuss the API decision here, this seemed logical but might not be the obvious to users. On the other hand, we don't want to hardcode this setting because other examples might want to enable the backward pass.

I turned off gradient storage (and thus the backwards pass) for all the robot examples with this change. We should discuss that for sure, but unless the backwards pass actually works, we can also save some memory. For sure it won't work for all the examples using MjWarp.

## Before your PR is "Ready for review"

- [x] Necessary tests have been added and new examples are tested (see `newton/tests/test_examples.py`)
- [x] Documentation is up-to-date
- [x] Code passes formatting and linting checks with `pre-commit run -a`

## Summary by CodeRabbit

- Refactor
  - Collision handling kernel is now generated at runtime and cached per instance, replacing direct kernel usage.
  - Collision pipeline updated to invoke the generated kernel.
  - Kernel configuration is determined during initialization based on runtime settings.
  - No user-facing behavior changes expected.

### CODE REVIEW

File: newton/examples/example_mujoco.py [hunk 1]
```diff
@@ -251,7 +251,7 @@ def __init__(
             builder = Example.create_model_builder(robot, num_envs, randomize, self.seed)
 
         # finalize model
-        self.model = builder.finalize()
+        self.model = builder.finalize(requires_grad=False)
```
Comment (eric-heiden) [inline newton/examples/example_mujoco.py]: `requires_grad` is False by default here, I think it would be better to not pass it explicitly in these examples then. The user might think they always have to do this when in reality gradient computation is something that needs to be _enabled_.

File: newton/_src/sim/collide.py [hunk 2]
```diff
@@ -135,6 +135,8 @@ def __init__(
         self.requires_grad = requires_grad
         self.edge_sdf_iter = edge_sdf_iter
 
+        self.handle_contact_pairs_kernel = generate_handle_contact_pairs_kernel(requires_grad)
```
Comment (eric-heiden) [inline newton/_src/sim/collide.py:138]: I like this approach!

File: newton/examples/example_mujoco.py (hunk 1 above)
Comment (adenzler-nvidia) [reply newton/examples/example_mujoco.py]: thanks for the hint, removed the explicit requires_grad from all the examples. The PR is quite trivial now :)

**Comment by adenzler-nvidia** [comment]
Should be ready to go now.

**Review by eric-heiden — APPROVED** [review]
